### PR TITLE
From for MatPolyRingZq with support for MatZ

### PR DESCRIPTION
**Pull request by Marcel583**
**Description**

We currently only have From for MatPolyRingZq with input MatPolyRingZq and  Into<ModulusPolynomialRingZq>. 
Since MatZ can be transformed into MatPolyRingZq, I changed our current implementation such that the input can be Into<MatPolyRingZq> and  Into<ModulusPolynomialRingZq>, without losing the efficiency of our current implementation.

This PR implements...
- [x] From for MatPolyRingZq with support for MatZ

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/integer_mod_q/mat_polynomial_ring_zq/from.rs [hunk 1]
```diff
@@ -32,66 +36,48 @@ impl<Mod: Into<ModulusPolynomialRingZq>> From<(&MatPolyOverZ, Mod)> for MatPolyn
...
     }
 }
 
-impl From<&MatPolynomialRingZq> for MatPolynomialRingZq {
-    /// Alias for [`MatPolynomialRingZq::clone`].
-    fn from(value: &MatPolynomialRingZq) -> Self {
-        value.clone()
-    }
-}
-
```
Comment (Marvin-Beckmann) [inline src/integer_mod_q/mat_polynomial_ring_zq/from.rs:84]: why do you remove this?

Comment (Marcel583) [reply src/integer_mod_q/mat_polynomial_ring_zq/from.rs]: This was a mistake, I did not want to remove it. I added it again in the next commit

**Review by Marvin-Beckmann — APPROVED** [review]
